dht_options.go

```diff
@@ -348,6 +350,14 @@ func DisableValues() Option {
 	}
 }
 
+// ProvidersOptions are options passed directly to the provider manager.
+func ProvidersOptions(opts []providers.Option) Option {
```

Comment:
might be nice to have this comment/name provide a clearer explanation and intuition of what this option does.